pkg/exec/v1alpha1/exec.go

```diff
@@ -0,0 +1,68 @@
+package v1alpha1
+
+import (
+	"os/exec"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"context"
+)
+
+// runner implements the Runner interface
+type runner struct {
+	timeout time.Duration
+	command string
+	args    []string
+}
+
+type builder struct {
+	runner *runner
+}
+
+// Builder returns instance of builder
+func Builder() *builder {
+	return &builder{runner: &runner{}}
+}
+
+// WithTimeout fill timeout field of runner struct
+func (b *builder) WithTimeout(timeout time.Duration) *builder {
+	b.runner.timeout = timeout
+	return b
+}
+
+// WithCommand fill command field of runner struct
+func (b *builder) WithCommand(cmd string) *builder {
+	b.runner.command = cmd
+	return b
+}
+
+// WithArgs fill args field of runner struct
+func (b *builder) WithArgs(args ...string) *builder {
+	b.runner.args = args
+	return b
+}
+
+// Build returns the instance of runner
+func (b *builder) Build() *runner {
+	return b.runner
+}
+
+// RunCommandWithTimeoutContext executes command provides and returns stdout
+// error. If command does not returns within given timout interval command will
+// be killed and return "Context time exceeded"
+func (r runner) RunCommandWithTimeoutContext() ([]byte, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), r.timeout)
+	defer cancel()
+
+	out, err := exec.CommandContext(ctx, r.command, r.args...).CombinedOutput()
```

Comment:
This error is generated because an external binary is being executed and can be ignored by invoking `gosec -exclude=G204` or by adding a comment `// nosec`, can be ignored.